### DISCUSSION TITLE
Add configuration  file for nicegui?

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1681] [Ideas / Feature Requests]
I found that the mkdocs.yml for mkdocs is very convient. For example [material navigation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/).we could config like this 
```
theme:
  features:
    - navigation.tabs
    - navigation.tabs.sticky
    -  navigation.sections
    - navigation.expand
```
which result in different kinds of menu.

**Comment by falkoschindler** [comment]
I feel like this would re-invent the wheel. We already have a language for configuring the UI: Python. By introducing a YAML configuration you need to duplicate features and documentation. At the moment I don't see the advantage. But maybe I'm missing the point and you have a certain use case in mind that is beyond what NiceGUI already offers.

**Reply by v479038280** [reply]
I post this because I think such kind of config file make some component e.g navigation very easy to use .

**Reply by rodja** [reply]
We designed NiceGUI to be a declarative UI like Flutter and SwiftUI because the flexibility of programming languages allows powerful interface design and component reuse. Defining UI in a config file is something which we really want to avoid in this project. But feel free to create another project on top of NiceGUI which uses config-files...
